libvast/test/detail/legacy_deserialize.cpp

```diff
@@ -0,0 +1,346 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#define SUITE legacy_deserialize
+
+#include "vast/detail/legacy_deserialize.hpp"
+
+#include "vast/as_bytes.hpp"
+#include "vast/concept/parseable/to.hpp"
+#include "vast/concept/parseable/vast/address.hpp"
+#include "vast/detail/serialize.hpp"
+#include "vast/factory.hpp"
+#include "vast/ids.hpp"
+#include "vast/legacy_type.hpp"
+#include "vast/qualified_record_field.hpp"
+#include "vast/synopsis.hpp"
+#include "vast/synopsis_factory.hpp"
+#include "vast/test/fixtures/actor_system.hpp"
+#include "vast/test/test.hpp"
+#include "vast/time_synopsis.hpp"
+
+#include <climits>
+#include <span>
+#include <string>
+#include <string_view>
+
+using namespace std::chrono_literals;
+using namespace vast;
+
+template <class... Ts>
+  requires(!std::is_rvalue_reference_v<Ts> && ...)
+bool ldes(std::vector<char>& buf, Ts&... xs) {
+  return detail::legacy_deserialize(buf, xs...);
+}
+
+TEST(deserialize_string) {
```

Comment:
The suite is already named `legacy_deserialize`, so please just name the test case `string`.

That allows for invoking `vast-test -s legacy_deserialize -t string` to run just this one test. The test case names only need to be unique within a single suite.

The same applies for all other test cases below.

Reply:
Test cases will have a shorter name after next push.